### PR TITLE
Fix: code quality issues

### PR DESCRIPTION
**Pull request by withshubh**
### Description
Hi :wave: I work at [DeepSource](https://deepsource.io), I ran DeepSource analysis on the forked copy of this repo and found some interesting [code quality issues](https://deepsource.io/gh/withshubh/dagster/issues/?category=recommended) in the codebase, opening this PR so you can assess if our platform is right and helpful for you.

### Motivation and Context
few fixes boost minor performance and few removes the anti-patterns in code.

### Summary of changes

- Removed unnecessary comprehension
- Used literal syntax instead of function calls to create data structure
- Used identity check for comparison with True
- Used `callable()` to check if the object is calllable
- Used `is` to compare type of objects
- Added .deepsource.toml to continuously analyze and detect code quality issues

### CODE REVIEW

File: .deepsource.toml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+version = 1
```
Comment (rexledesma) [inline .deepsource.toml:1]: This file should be deleted

**Review by rexledesma — CHANGES_REQUESTED** [review]
Hi @withshubh - thanks for the PR. The changes look great, but we're not considering an integration with deepsource at this time. If you remove the config file, we can merge this in.

**Comment by withshubh** [comment]
@rexledesma deleted the file. :sparkles:

**Comment by withshubh** [comment]
Hi @rexledesma Please review the PR

File: python_modules/libraries/dagster-aws/dagster_aws/utils/mrjob/retry.py [hunk 2]
```diff
@@ -101,7 +101,7 @@ def __getattr__(self, name):
 
         if name in self.__unwrap_methods:
             return partial(x.__func__, self)
-        elif hasattr(x, "__call__"):
```
Comment (mgasner) [inline python_modules/libraries/dagster-aws/dagster_aws/utils/mrjob/retry.py:104]: this is vendored code so i would hesitate to make random changes to it

Comment (withshubh) [reply python_modules/libraries/dagster-aws/dagster_aws/utils/mrjob/retry.py]: @mgasner removed the fix!

**Comment by rexledesma** [comment]
On a second look, this is mostly aesthetic changes that don't really affect the performance of the codebase. Since we're not considering an integration with deepsource, I am closing this PR.

**Pull request closed without merging**